File: easy/0541_reverse_string_ii/solution.cpp
#include <algorithm>
#include <string>

class Solution {
public:
  std::string reverseStr(std::string s, int k) {
    for (int i = 0; i < s.size(); i += 2 * k)
      reverse(begin(s) + i, begin(s) + std::min(i + k, (int)s.size()));

    return s;
  }
};

File: easy/0541_reverse_string_ii/solution.py
class Solution:
    def reverseStr(self, s: str, k: int) -> str:
        s = list(s)

        start = 0
        end = min(k - 1, len(s) - 1)

        while True:
            left = start
            right = end

            while left < right:
                s[left], s[right] = s[right], s[left]
                left += 1
                right -= 1

            start = start + 2 * k
            end = min(start + k - 1, len(s) - 1)

            if start >= len(s):
                break

        return "".join(s)
